assets/cms/js/vue/Manager.js

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable no-new */
+/* global Vue */
+import Context from './Context'
+
+class Manager {
+    /**
+     * Ensures that our Concrete.Vue manager is available on the window object.
+     * Note: Do NOT call this before the global Concrete object is created in the CMS context.
+     *
+     * @param window
+     */
+    static bindToWindow(window) {
+        if (!window.Concrete) {
+            window.Concrete = {}
+        }
+        if (!window.Concrete.Vue) {
+            window.Concrete.Vue = new Manager()
+            window.Concrete.Vue.contexts = []
+        }
+    }
+
+    /**
+     * Returns the Vue Context object for the current string `context`
+     *
+     * @param context
+     * @returns {*}
+     */
+    getContext(context) {
+        return window.Concrete.Vue.contexts.filter(obj => obj.context === context)[0]
+    }
+
+    /**
+     * Actives a particular context (and its components) for a particular selector.
+     *
+     * @param context
+     * @param selector
+     * @param parent
+     */
+    activateContext(context, selector, parent) {
+        var contextObject = this.getContext(context)
+        var selectors = parent.querySelectorAll(selector)
+        selectors.forEach(function (element) {
+            new Vue({
```

Comment:
The final object that's build here should be exposed to the caller somehow before being inflated into a new vue context, a good way would be a callback function:
```
.activateContext('some-context', 'selector', function(options) {
    return {
        ...options,
        theRestOfMyStuff: {...}
    }
});
```
and this code would do:
```
new Vue(callback({
    el: selector,
    components: components,
}));
```
This is really important because this top level vue context will need to be configured with things like data, methods, and everything else a component could use

Reply:
What if we have more than one Vue "app", each one with its own set of data? I mean, wouldn't it be easier to strip out `selector` and `parent`, and just pass the callback `Vue` and `components`, so that the callback could do whatever it wants (including for example assigning the Vue instance to a variable - which is impossible with this approach)

Reply:
Yeah definitely, the ideal implementation would be:
```
    activateContext(context, callback) {
        return callback(Vue, {
            components: this.getContext(context)
        })
    }
```